### PR TITLE
Use custom bison with patch to fix macOS High Sierra build

### PR DESCRIPTION
**Pull request by isaachier**
While travis checks this builds correctly, can you let me know where (if anywhere) is a good place to post a patched tarball like the one currently at https://github.com/isaachier/bison/files/1471873/bison-3.0.4.tar.gz?

### CODE REVIEW

**Comment by isaachier** [comment]
Fixes #1196

File: cmake/projects/bison/hunter.cmake [hunk 1]
```diff
@@ -6,14 +6,21 @@ include(hunter_download)
...
+  URL "https://ftp.gnu.org/gnu/bison/bison-3.0.4.tar.gz"
+  SHA1 "ec1f2706a7cfedda06d29dc394b03e092a1e1b74"
+)
+
+hunter_add_version(
+  PACKAGE_NAME
+  bison
+  VERSION
+  "3.0.4-p0"
+  URL "https://github.com/isaachier/bison/files/1471873/bison-3.0.4.tar.gz"
```
Comment (ruslo) [inline cmake/projects/bison/hunter.cmake]: I've created repo in `hunter-packages`: https://github.com/hunter-packages/bison

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2812
* https://travis-ci.org/ingenue/hunter/builds/302444442?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2813
* https://travis-ci.org/ingenue/hunter/builds/302459633?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
> Uploading

It's not cacheable, I forgot :)

Released: https://github.com/ruslo/hunter/releases/tag/v0.19.157

**Pull request merged**